colorbleed/plugins/maya/publish/validate_animation_out_set_related_node_ids.py

```diff
@@ -3,87 +3,127 @@
+    shapes will need to get the correct ID to ensure the look assignment still
+    works on the new shape.
 
     """
 
     order = colorbleed.api.ValidateContentsOrder
-    families = ['colorbleed.animation']
+    families = ['colorbleed.animation', "colorbleed.pointcache"]
     hosts = ['maya']
     label = 'Animation Out Set Related Node Ids'
-    actions = [colorbleed.api.SelectInvalidAction]
+    actions = [colorbleed.api.SelectInvalidAction, colorbleed.api.RepairAction]
     optional = True
 
     ignore_types = ("constraint",)
 
     def process(self, instance):
         """Process all meshes"""
 
-        # Ensure all nodes have a cbId
+        # Ensure all nodes have a cbId and a related ID to the original shapes
+        # if a deformer has been created on the shape
         invalid = self.get_invalid(instance)
         if invalid:
             raise RuntimeError("Nodes found with non-related "
                                "asset IDs: {0}".format(invalid))
 
     @classmethod
-    def get_pointcache_nodes(cls, instance):
-
-        # Get out_SET
-        sets = cmds.ls(instance, type='objectSet')
-        pointcache_sets = [x for x in sets if x == 'out_SET']
-
-        nodes = list()
-        for s in pointcache_sets:
-            members = cmds.sets(s, query=True)
-            members = cmds.ls(members, long=True)  # ensure long names
-            descendents = cmds.listRelatives(members,
-                                             allDescendents=True,
-                                             fullPath=True) or []
-            descendents = cmds.ls(descendents,
-                                  noIntermediate=True,
-                                  long=True)
-            hierarchy = members + descendents
-            nodes.extend(hierarchy)
-
-        # ignore certain node types (e.g. constraints)
-        ignore = cmds.ls(nodes, type=cls.ignore_types, long=True)
-        if ignore:
-            ignore = set(ignore)
-            nodes = [node for node in nodes if node not in ignore]
-
-        return nodes
-
-    @classmethod
     def get_invalid(cls, instance):
-        invalid_items = []
+        """Get all nodes which do not match the criteria"""
+
+        invalid = []
 
         # get asset id
-        nodes = cls.get_pointcache_nodes(instance)
+        nodes = instance.data.get("out_hierarchy", instance[:])
         for node in nodes:
+
+            # We only check when the node is *not* referenced
+            if cmds.referenceQuery(node, isNodeReferenced=True):
+                continue
+
+            # Check if node is a shape as deformers only work on shapes
+            obj_type = cmds.objectType(node, isAType="shape")
+            if not obj_type:
+                continue
+
+            # Get the current id of the node
             node_id = lib.get_id(node)
-            if not node_id:
-                invalid_items.append(node)
+            if node_id:
```

Comment:
I don't believe this is what you want @aardschok . Should this be ìf not node_id``?